db/batchlog_manager.cc

```diff
@@ -115,10 +115,12 @@ future<> db::batchlog_manager::start() {
 }
 
 future<> db::batchlog_manager::drain() {
-    blogger.info("Asked to drain");
-    if (!_stop.abort_requested()) {
-        _stop.request_abort();
+    if (_stop.abort_requested()) {
+        co_return;
```

Comment:
Shouldn't it join with the other caller? Otherwise we can return before the drain is complete.

Reply:
ok. It's not worse than before anyway.